hudi-client/hudi-client-common/src/test/java/org/apache/hudi/client/transaction/FileSystemBasedLockProviderTestClass.java

```diff
@@ -76,7 +76,7 @@ public boolean tryLock(long time, TimeUnit unit) {
     try {
       int numRetries = 0;
       while (fs.exists(new Path(lockPath + "/" + LOCK_NAME))
-          && (numRetries <= lockConfiguration.getConfig().getInteger(LOCK_ACQUIRE_NUM_RETRIES_PROP_KEY))) {
+          && (numRetries++ <= lockConfiguration.getConfig().getInteger(LOCK_ACQUIRE_NUM_RETRIES_PROP_KEY))) {
```

Comment:
This will non fix the test flakiness. Test assumes a different exception to be raised.

Reply:
sure. i will let you fix the actual test. but this fix seems inevitable for now.